R/zchunk_LA126.distribution.R

```diff
@@ -36,123 +40,180 @@ module_energy_LA126.distribution_DISABLED <- function(command, ...) {
     all_data <- list(...)[[1]]
 
     # Load required inputs
-    calibrated_techs <- get_data(all_data, "energy/calibrated_techs")
-    L1011.en_bal_EJ_R_Si_Fi_Yh <- get_data(all_data, "temp-data-inject/L1011.en_bal_EJ_R_Si_Fi_Yh")
-    L122.out_EJ_R_gasproc_F_Yh <- get_data(all_data, "temp-data-inject/L122.out_EJ_R_gasproc_F_Yh")
+    get_data(all_data, "temp-data-inject/L1011.en_bal_EJ_R_Si_Fi_Yh") %>%
+      gather(year, value, -GCAM_region_ID, -sector, -fuel) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L1011.en_bal_EJ_R_Si_Fi_Yh
+
+    get_data(all_data, "temp-data-inject/L122.out_EJ_R_gasproc_F_Yh") %>%
+      gather(year, value, -GCAM_region_ID, -sector, -fuel) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L122.out_EJ_R_gasproc_F_Yh
+
     L123.out_EJ_R_elec_F_Yh <- get_data(all_data, "L123.out_EJ_R_elec_F_Yh")
     L123.out_EJ_R_indchp_F_Yh <- get_data(all_data, "L123.out_EJ_R_indchp_F_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # ===================================================
+    # ELECTRICITY OWNUSE, i.e., electricity consumed onsite prior to any transmission and distribution losses
+    # Summing industrial CHP electricity generation and electricity generation by GCAM region ID and year
+    # Summing electricity generation by GCAM region ID and year
+    L123.out_EJ_R_elec_F_Yh %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value_elec_F_Yh = sum(value)) ->
+      a
+
+    # Summing industrial CHP electricity generation and then joining it with electricity generation
+    L123.out_EJ_R_indchp_F_Yh %>%
+      group_by(GCAM_region_ID, year) %>%
+      summarise(value_indchp_F_Yh = sum(value)) %>%
+      left_join_error_no_match(a, by = c("GCAM_region_ID", "year")) %>%
+      mutate(value_b = value_elec_F_Yh + value_indchp_F_Yh) %>% # Summing electricity generation and industrial CHP electricity generation. This creates values for table, L126.in_EJ_R_elecownuse_F_Yh (i.e., input).
+      select(-value_elec_F_Yh, -value_indchp_F_Yh) ->
+      b
+
+    # Filtering energy balance by "net_electricity ownuse" sector and joining with previous table
+    L1011.en_bal_EJ_R_Si_Fi_Yh %>%
+      filter(sector == "net_electricity ownuse") %>%
+      mutate(sector = substr(sector, 5, 30)) %>% # Renaming sector as "electricity ownuse"
+      group_by(GCAM_region_ID, sector, fuel, year) %>%
```

Comment:
please note that I didn't know how to do the reverse, so I used mutate/replace:
mutate(sector = replace(sector, sector == "net_gas pipeline", "gas pipeline"))